### ISSUE TITLE
[Support]: recording skips 50+ seconds during recording via manual event creation

### ISSUE THREAD

**Issue by HWL1968** [blakeblackshear/frigate#9229]
### Describe the problem you are having

Hi all,
First: I love Frigate in the 0.13 version. Works better and more stable than the previous version. Especially the manual event creation is awesome. Many thx!!
This brings me to my issue:
System:
- running Frigate 0.13.0-rc1 in docker on a NAS
- camera's: 2 foscam 9900

I use manual events to trigger and stop recording. I do this by:
- trigger by a motion sensor (I use NodeRed to start / stop the recording)
- then generate a POST to frigate: url="http://192.168.x.xxx:5000/api/events/Voordeur/SENSOR/create" and payload ="{ "sub_label": "testing",  "duration": "null", "include_recording": true }" and response={"event_id":"1704539887.745733-6ki34s","message":"Successfully created event.","success":true}
- after a duration (delay) of 2 minutes I PUT "http://192.168.1.11:5000/api/events/1704539887.745733-6ki34s/end" with response "{"message":"Event successfully ended.","success":true}"

Then the weird point is: the recording starts and stops as wanted but during the recording somewhere around 50 to 60 seconds (not always at the same moment) the recording stops and starts again around 30-60 seconds later. Playback doesn't stop but jumps the stopped period. (so keeps playing as if it didn't stop but misses the mentioned timeframe)

Any idea what could be wrong? Something in a setting I missed?

Best regards and many thanks.

### Version

0.13.0-49814B3

### Frigate config file

```yaml
ui:
  # Optional: Set a timezone to use in the UI (default: use browser local time)
  timezone: Europe/Amsterdam

mqtt:
  host: 192.168.x.xxx
  port: 11883
  user: frigate
  password: XXXXXXX
  topic_prefix: frigate
  client_id: frigate

birdseye:
  enabled: True
  width: 1920
  height: 1080
  quality: 8
  mode: continuous

record:
  # Optional: Enable recording (default: shown below)
  enabled: True
  # Optional: Number of days to retain recordings regardless of events (default: shown below)
  # NOTE: This should be set to 0 and retention should be defined in events section below
  #       if you only want to retain recordings of events.
  retain:
    days: 0
  # Optional: Event recording settings
  events:
    # Optional: Number of seconds before the event to include (default: shown below)
    pre_capture: 2
    # Optional: Number of seconds after the event to include (default: shown below)
    post_capture: 8
    # Optional: Objects to save recordings for. (default: all tracked objects)
    objects:
      - person
    # Optional: Restrict recordings to objects that entered any of the listed zones (default: no required zones)
    required_zones: []
    # Optional: Retention settings for recordings of events
    retain:
      # Required: Default retention days (default: shown below)
      default: 20

# Optional: Configuration for the jpg snapshots written to the clips directory for each event
# NOTE: Can be overridden at the camera level
snapshots:
  # Optional: Enable writing jpg snapshot to /media/frigate/clips (default: shown below)
  # This value can be set via MQTT and will be updated in startup based on retained value
  enabled: True
  # Optional: print a timestamp on the snapshots (default: shown below)
  timestamp: True
  # Optional: draw bounding box on the snapshots (default: shown below)
  bounding_box: True
  # Optional: crop the snapshot (default: shown below)
  crop: False
  # Optional: height to resize the snapshot to (default: original size)
  #height: 175
  # Optional: Restrict snapshots to objects that entered any of the listed zones (default: no required zones)
  required_zones: []
  # Optional: Camera override for retention settings (default: global values)
  retain:
    # Required: Default retention days (default: shown below)
    default: 20

cameras:
  Voordeur:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://xxxx:xxxx@192.168.1.31:88/videoMain
          roles:
            - detect
            - record
            - rtmp
    motion:
      improve_contrast: True
      threshold: 1
  # Optional: Minimum size in pixels in the resized motion image that counts as motion (default: shown below)
  # Increasing this value will prevent smaller areas of motion from being detected. Decreasing will
  # make motion detection more sensitive to smaller moving objects.
  # As a rule of thumb:
  #  - 10 - high sensitivity
  #  - 30 - medium sensitivity
  #  - 50 - low sensitivity
      contour_area: 10
      mask:
        - 1920,1080,1920,0,0,0,0,42,198,47,209,143,491,168,807,438,707,517,1384,1080
    objects:
      track:
      - person
      filters:
        person:
          mask:
            - 1920,22,1920,871,1898,1080,831,347,619,262,611,0
    detect:    
      width: 1920
      height: 1080
      fps: 5
      enabled: True
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
      retain:
        default: 20
    mqtt:
      # Optional: Enable publishing snapshot via mqtt for camera (default: shown below)
      # NOTE: Only applies to publishing image data to MQTT via 'frigate/<camera_name>/<object_name>/snapshot'.
      # All other messages will still be published.
      enabled: True
  Zijkant:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://xxxx:xxxx@192.168.1.32:88/videoMain
          roles:
            - detect
            - record
            - rtmp
    motion:
      improve_contrast: True
      threshold: 1
  # Optional: Minimum size in pixels in the resized motion image that counts as motion (default: shown below)
  # Increasing this value will prevent smaller areas of motion from being detected. Decreasing will
  # make motion detection more sensitive to smaller moving objects.
  # As a rule of thumb:
  #  - 10 - high sensitivity
  #  - 30 - medium sensitivity
  #  - 50 - low sensitivity
      contour_area: 10
      mask:
        - 1920,46,1920,0,0,0,0,88,662,55,824,467,1351,525,1643,53
    objects:
      track:
      - person
      filters:
        person:
          mask:
            - 835,435,1468,24,1449,0,0,0,0,63,652,64
    detect:    
      width: 1920
      height: 1080
      fps: 5
      enabled: True
    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
      retain:
        default: 20
    mqtt:
      # Optional: Enable publishing snapshot via mqtt for camera (default: shown below)
      # NOTE: Only applies to publishing image data to MQTT via 'frigate/<camera_name>/<object_name>/snapshot'.
      # All other messages will still be published.
      enabled: True
```

### Relevant log output

```shell
(none found, only loggings regarding the other camera)
```

### FFprobe output from your camera

```shell
FFPROBE OUTPUT
Stream 0:
Return Code: 0

Video:

Codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Resolution: 1920x1080
FPS: 155/6

Audio:

Codec: PCM mu-law / G.711 mu-law
```

### Frigate stats

```json
{
    "cameras": {
        "Voordeur": {
            "audio_dBFS": 0,
            "audio_rms": 0,
            "camera_fps": 5.1,
            "capture_pid": 326,
            "detection_enabled": 1,
            "detection_fps": 0.2,
            "ffmpeg_pid": 5312,
            "pid": 323,
            "process_fps": 0.1,
            "skipped_fps": 0
        },
        "Zijkant": {
            "audio_dBFS": 0,
            "audio_rms": 0,
            "camera_fps": 5.1,
            "capture_pid": 329,
            "detection_enabled": 1,
            "detection_fps": 0,
            "ffmpeg_pid": 335,
            "pid": 324,
            "process_fps": 5.1,
            "skipped_fps": 0
        }
    },
    "cpu_usages": {
        "1": {
            "cmdline": "/package/admin/s6/command/s6-svscan -d4 -- /run/service",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "14": {
            "cmdline": "s6-supervise s6-linux-init-shutdownd",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "16": {
            "cmdline": "/package/admin/s6-linux-init/command/s6-linux-init-shutdownd -c /run/s6/basedir -g 3000 -C -B",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "23": {
            "cmdline": "s6-supervise frigate",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "24": {
            "cmdline": "s6-supervise go2rtc-healthcheck",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "25": {
            "cmdline": "s6-supervise s6rc-oneshot-runner",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "26": {
            "cmdline": "s6-supervise nginx-log",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "27": {
            "cmdline": "s6-supervise nginx",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "28": {
            "cmdline": "s6-supervise frigate-log",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "29": {
            "cmdline": "s6-supervise go2rtc",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "30": {
            "cmdline": "s6-supervise s6rc-fdholder",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "31": {
            "cmdline": "s6-supervise go2rtc-log",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "40": {
            "cmdline": "/package/admin/s6-2.11.3.2/command/s6-fdholderd -1 -i data/rules",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "41": {
            "cmdline": "/package/admin/s6/command/s6-ipcserverd -1 -- /package/admin/s6/command/s6-ipcserver-access -v0 -E -l0 -i data/rules -- /package/admin/s6/command/s6-sudod -t 30000 -- /package/admin/s6-rc/command/s6-rc-oneshot-run -l ../.. --",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "79": {
            "cmdline": "s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/go2rtc",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "80": {
            "cmdline": "s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/frigate",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "81": {
            "cmdline": "s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/nginx",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "87": {
            "cmdline": "/usr/local/go2rtc/bin/go2rtc -config=/dev/shm/go2rtc.yaml",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.1"
        },
        "105": {
            "cmdline": "bash ./run.user go2rtc-healthcheck",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "108": {
            "cmdline": "python3 -u -m frigate",
            "cpu": "11.0",
            "cpu_average": "2",
            "mem": "2.1"
        },
        "121": {
            "cmdline": "nginx: master process nginx",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.1"
        },
        "145": {
            "cmdline": "nginx: worker process",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.1"
        },
        "146": {
            "cmdline": "nginx: worker process",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.1"
        },
        "147": {
            "cmdline": "nginx: worker process",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "155": {
            "cmdline": "nginx: worker process",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "164": {
            "cmdline": "nginx: cache manager process",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        },
        "295": {
            "cmdline": "frigate.logger       ",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.6"
        },
        "297": {
            "cmdline": "frigate.recording_manager",
            "cpu": "0.0",
            "cpu_average": "1",
            "mem": "0.9"
        },
        "306": {
            "cmdline": "/usr/bin/python3 -c from multiprocessing.resource_tracker import main;main(45)",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.1"
        },
        "307": {
            "cmdline": "frigate.detector.cpu ",
            "cpu": "0.0",
            "cpu_average": "24",
            "mem": "0.8"
        },
        "309": {
            "cmdline": "frigate.output       ",
            "cpu": "0.7",
            "cpu_average": "1",
            "mem": "0.8"
        },
        "312": {
            "cmdline": "ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 1920x1080 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.1"
        },
        "313": {
            "cmdline": "ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 1920x1080 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.1"
        },
        "314": {
            "cmdline": "ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 1920x1080 -i pipe: -f mpegts -s 1920x1080 -codec:v mpeg1video -q 8 -bf 0 pipe:",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.1"
        },
        "323": {
            "cmdline": "frigate.process:Voordeur",
            "cpu": "0.0",
            "cpu_average": "1",
            "mem": "1.0"
        },
        "324": {
            "cmdline": "frigate.process:Zijkant",
            "cpu": "3.7",
            "cpu_average": "3",
            "mem": "0.8"
        },
        "326": {
            "cmdline": "frigate.capture:Voordeur",
            "cpu": "0.0",
            "cpu_average": "2",
            "mem": "0.7"
        },
        "329": {
            "cmdline": "frigate.capture:Zijkant",
            "cpu": "6.6",
            "cpu_average": "6",
            "mem": "0.7"
        },
        "335": {
            "cmdline": "ffmpeg -hide_banner -loglevel warning -threads 2 -hwaccel_flags allow_profile_mismatch -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format vaapi -user_agent FFmpeg Frigate/0.13.0-49814b3 -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://admin:Hu63rtu5%21@192.168.1.32:88/videoMain -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an /tmp/cache/Zijkant@%Y%m%d%H%M%S%z.mp4 -r 5 -vf fps=5,scale_vaapi=w=1920:h=1080:format=nv12,hwdownload,format=nv12,format=yuv420p -threads 2 -f rawvideo -pix_fmt yuv420p pipe:",
            "cpu": "8.8",
            "cpu_average": "10",
            "mem": "0.5"
        },
        "5312": {
            "cmdline": "ffmpeg -hide_banner -loglevel warning -threads 2 -hwaccel_flags allow_profile_mismatch -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format vaapi -user_agent FFmpeg Frigate/0.13.0-49814b3 -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://admin:Hu63rtu5%21@192.168.1.31:88/videoMain -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an /tmp/cache/Voordeur@%Y%m%d%H%M%S%z.mp4 -r 5 -vf fps=5,scale_vaapi=w=1920:h=1080:format=nv12,hwdownload,format=nv12,format=yuv420p -threads 2 -f rawvideo -pix_fmt yuv420p pipe:",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.2"
        },
        "5321": {
            "cmdline": "sleep 30s",
            "cpu": "0.0",
            "cpu_average": "0",
            "mem": "0.0"
        }
    },
    "detection_fps": 0.2,
    "detectors": {
        "cpu": {
            "detection_start": 0,
            "inference_speed": 378.68,
            "pid": 307
        }
    },
    "gpu_usages": {
        "intel-vaapi": {
            "gpu": "4.82%",
            "mem": "-%"
        }
    },
    "processes": {
        "go2rtc": {
            "pid": 87
        },
        "logger": {
            "pid": 295
        },
        "recording": {
            "pid": 297
        }
    },
    "service": {
        "last_updated": 1704540948,
        "latest_version": "0.12.1",
        "storage": {
            "/dev/shm": {
                "free": 88.4,
                "mount_type": "tmpfs",
                "total": 96,
                "used": 7.6
            },
            "/media/frigate/clips": {
                "free": 3606430.4,
                "mount_type": "ext4",
                "total": 11342624.2,
                "used": 7733105.8
            },
            "/media/frigate/recordings": {
                "free": 3606430.4,
                "mount_type": "ext4",
                "total": 11342624.2,
                "used": 7733105.8
            },
            "/tmp/cache": {
                "free": 950.3,
                "mount_type": "tmpfs",
                "total": 953.7,
                "used": 3.4
            }
        },
        "temperatures": {},
        "uptime": 3353,
        "version": "0.13.0-49814b3"
    }
}
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

foscam 9900

**Comment by NickM-27** [comment]
this is because manual event recordings still adhere to the recording retention. In this case you only have recordings enabled for events and the default retention mode is motion so any recording segment that doesn't have motion will not be retained

**Comment by HWL1968** [comment]
Hi Nick,
Thx for your clarification. I thought the manual (MQTT) event would last until the 'stop' command. And yes (before you answer: read the manual, I did :-)  but since the mqtt is not yet official I thought that would be handled different from events). 
Reason: it is stated that you can give a duration for the recording, which gave the same result (as in: skipping in the recording). Therefor I switched toward a start / stop sequence.

Just that I understand it correctly: MQTT starting of a recording is handled as a 'standard' event. Thus by defining the 'record' and subsequently 'events' I would set the behaviour for all events including the MQTT start/stop?

How would I have to change my config to make sure it keeps the complete recorded sequence? My config right now is:

```
record:
  retain:
    days: 0
  events:
    retain:
      default: 20
```

Would the following be correct?
```
record:
  retain:
    days: 0
    **_mode: all_**
  events:
    retain:
      default: 20
    **_mode: all_**
```

Or would it be necessary to put it only under 'record -> retain' or (since it is only appropriate for an event) only under 'events:'?

Thanks again.

**Comment by HWL1968** [comment]
Hi Nick,

For the record: I got it fixed. The body of my request wasn't correct. It was a text but not as JSON (appeared as JSON as it was made readable during the debugging).
Now it is working and the duration is as expected without any frames missing. 

This is besides the fixes I made thanks to your input.

For future reference:
- make sure your config is as described (record: -> events: -> retain: -> mode: all )
- in NodeRed use the JSON building block to convert the payload to actual JSON (and not just text as I did). Then you will be able to manual (or via MQTT as I did) start and stop the recording.

Best regards.
